samples/embedded-map/src/components/EmbeddedMap/EmbeddedMapModel.ts

```diff
@@ -6,49 +6,69 @@ import {
 } from "@vertigis/web/models";
 import { throttle } from "@vertigis/web/ui";
 import Point from "esri/geometry/Point";
+import { whenDefinedOnce } from "esri/core/watchUtils";
 import { Viewer } from "mapillary-js";
 
 /**
  *  Convert Mapillary bearing to a Scene's camera rotation.
  *  @param bearing Mapillary bearing in degrees (degrees relative to due north).
  *  @returns Scene camera rotation in degrees (degrees rotation of due north).
  */
-
 function getCameraRotationFromBearing(bearing: number): number {
     return 360 - bearing;
 }
 
+interface MapillaryCamera {
+    latitude: number;
+    longitude: number;
+    heading: number;
+    tilt: number;
+    fov: number;
+}
+
+interface GcxMap extends MapExtension {
+    view: __esri.SceneView;
+}
+
 @serializable
 export default class EmbeddedMapModel extends ComponentModelBase {
-    private _map: MapExtension | undefined;
-    get map(): MapExtension | undefined {
-        return this._map;
-    }
-    @importModel("map-extension")
-    set map(instance: MapExtension | undefined) {
-        if (instance === this._map) {
-            return;
-        }
+    // For demonstration purposes only.
+    // Replace this with your own client ID from mapillary.com
+    readonly mapillaryKey =
+        "ZU5PcllvUTJIX24wOW9LSkR4dlE5UTo3NTZiMzY4ZjBlM2U2Nzlm";
 
-        // If an instance already exists, clean it up first.
-        if (this._map) {
-            // eslint-disable-next-line @typescript-eslint/no-floating-promises
-            this._unsyncMaps();
-        }
+    readonly imageQueryUrl = "https://a.mapillary.com/v3/images";
+    readonly searchRadius = 500; // meters
 
-        this._map = instance;
+    private _currentPosition: { latitude: number; longitude: number };
+    private _lastMarkerUpdate: any;
+    private _updating = false;
+    private _viewerUpdateHandle: IHandle;
 
-        // A new instance is being set - sync the map.
-        if (instance) {
-            // eslint-disable-next-line @typescript-eslint/no-floating-promises
-            this._syncMaps();
+    mouseDownHandler = (): void => (this._lastMarkerUpdate = undefined);
+    mouseUpHandler = (): void => {
+        if (
+            this.mapillary?.isNavigable &&
+            !this._updating &&
+            this._lastMarkerUpdate
+        ) {
+            this._updating = true;
+
+            const { latitude, longitude } = this._lastMarkerUpdate
+                .geometry as __esri.geometry.Point;
+            this._lastMarkerUpdate = undefined;
+
+            void this._moveCloseToPosition(latitude, longitude);
         }
-    }
+    };
 
-    // For demonstration purposes only.
-    // Replace this with your own client ID from mapillary.com
-    readonly mapillaryKey =
-        "ZU5PcllvUTJIX24wOW9LSkR4dlE5UTo3NTZiMzY4ZjBlM2U2Nzlm";
+    private _syncGcxMap = true;
+    get sync(): boolean {
```

Comment:
The getter/setter doesn't seem to be doing much other than forcing the value to be `true` or `false`. Is that necessary?

Reply:
Nope :)